### PR TITLE
Move all Api...Service classes to Sandbox package

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
This is done to restrict the usage of NamedLoggerFactory and most usages of MetricsRegistry
to a single Bazel package, thus allowing a possible replacement to not have to become a
dependency of multiple packages.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/tracking/TrackerMap.scala [hunk 1]
```diff
@@ -93,24 +91,25 @@ class TrackerMap(retentionPeriod: FiniteDuration, loggerFactory: NamedLoggerFact
 }
 
 object TrackerMap {
-  case class Key(application: String, party: String)
+
+  final case class Key(application: String, party: String)
+
+  sealed trait AsyncResourceState[+T <: AutoCloseable]
```
Comment (rautenrieth-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/tracking/TrackerMap.scala]: Why did this trait become generic?

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/tracking/TrackerMap.scala]: It used to be part of `AsyncResource` and this "locked" the generic `T` correctly but caused a warning that was silenced. I did this to make sure `AsyncResource` can use it by explicitly assigning the type but without causing the warning (caused by [this](https://github.com/scala/bug/issues/4440)).

**Pull request merged**